File: leetcode392.go
package main

func isSubsequence(s string, t string) bool {
	if len(s) == 0 {
		return true
	}
	for i, j := 0, 0; j < len(t); j++ {
		if s[i] == t[j] {
			i++
		}
		if i == len(s) {
			return true
		}

	}
	return false
}

File: leetcode392.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isSubsequence = function(s, t) {
    let i = 0;
    let j = 0;
    while(i < s.length && j < t.length) {
        if(s[i] === t[j]) {
            i++;
        }
        j++;
    }
    return i === s.length;
};